src/main/java/com/pokegoapi/api/pokemon/PokemonCpUtils.java

```diff
@@ -0,0 +1,247 @@
+		CPMULTIPLIER_LEVEL.put(0.387592406f, 8.5f);
+		CPMULTIPLIER_LEVEL.put(0.39956728f, 9f);
+		CPMULTIPLIER_LEVEL.put(0.411193551f, 9.5f);
+		CPMULTIPLIER_LEVEL.put(0.42250001f, 10f);
+		CPMULTIPLIER_LEVEL.put(0.432926419f, 10.5f);
+		CPMULTIPLIER_LEVEL.put(0.44310755f, 11f);
+		CPMULTIPLIER_LEVEL.put(0.453059958f, 11.5f);
+		CPMULTIPLIER_LEVEL.put(0.46279839f, 12f);
+		CPMULTIPLIER_LEVEL.put(0.472336083f, 12.5f);
+		CPMULTIPLIER_LEVEL.put(0.48168495f, 13f);
+		CPMULTIPLIER_LEVEL.put(0.4908558f, 13.5f);
+		CPMULTIPLIER_LEVEL.put(0.49985844f, 14f);
+		CPMULTIPLIER_LEVEL.put(0.508701765f, 14.5f);
+		CPMULTIPLIER_LEVEL.put(0.51739395f, 15f);
+		CPMULTIPLIER_LEVEL.put(0.525942511f, 15.5f);
+		CPMULTIPLIER_LEVEL.put(0.53435433f, 16f);
+		CPMULTIPLIER_LEVEL.put(0.542635767f, 16.5f);
+		CPMULTIPLIER_LEVEL.put(0.55079269f, 17f);
+		CPMULTIPLIER_LEVEL.put(0.558830576f, 17.5f);
+		CPMULTIPLIER_LEVEL.put(0.56675452f, 18f);
+		CPMULTIPLIER_LEVEL.put(0.574569153f, 18.5f);
+		CPMULTIPLIER_LEVEL.put(0.58227891f, 19f);
+		CPMULTIPLIER_LEVEL.put(0.589887917f, 19.5f);
+		CPMULTIPLIER_LEVEL.put(0.59740001f, 20f);
+		CPMULTIPLIER_LEVEL.put(0.604818814f, 20.5f);
+		CPMULTIPLIER_LEVEL.put(0.61215729f, 21f);
+		CPMULTIPLIER_LEVEL.put(0.619399365f, 21.5f);
+		CPMULTIPLIER_LEVEL.put(0.62656713f, 22f);
+		CPMULTIPLIER_LEVEL.put(0.633644533f, 22.5f);
+		CPMULTIPLIER_LEVEL.put(0.64065295f, 23f);
+		CPMULTIPLIER_LEVEL.put(0.647576426f, 23.5f);
+		CPMULTIPLIER_LEVEL.put(0.65443563f, 24f);
+		CPMULTIPLIER_LEVEL.put(0.661214806f, 24.5f);
+		CPMULTIPLIER_LEVEL.put(0.667934f, 25f);
+		CPMULTIPLIER_LEVEL.put(0.674577537f, 25.5f);
+		CPMULTIPLIER_LEVEL.put(0.68116492f, 26f);
+		CPMULTIPLIER_LEVEL.put(0.687680648f, 26.5f);
+		CPMULTIPLIER_LEVEL.put(0.69414365f, 27f);
+		CPMULTIPLIER_LEVEL.put(0.700538673f, 27.5f);
+		CPMULTIPLIER_LEVEL.put(0.70688421f, 28f);
+		CPMULTIPLIER_LEVEL.put(0.713164996f, 28.5f);
+		CPMULTIPLIER_LEVEL.put(0.71939909f, 29f);
+		CPMULTIPLIER_LEVEL.put(0.725571552f, 29.5f);
+		CPMULTIPLIER_LEVEL.put(0.7317f, 30f);
+		CPMULTIPLIER_LEVEL.put(0.734741009f, 30.5f);
+		CPMULTIPLIER_LEVEL.put(0.73776948f, 31f);
+		CPMULTIPLIER_LEVEL.put(0.740785574f, 31.5f);
+		CPMULTIPLIER_LEVEL.put(0.74378943f, 32f);
+		CPMULTIPLIER_LEVEL.put(0.746781211f, 32.5f);
+		CPMULTIPLIER_LEVEL.put(0.74976104f, 33f);
+		CPMULTIPLIER_LEVEL.put(0.752729087f, 33.5f);
+		CPMULTIPLIER_LEVEL.put(0.75568551f, 34f);
+		CPMULTIPLIER_LEVEL.put(0.758630378f, 34.5f);
+		CPMULTIPLIER_LEVEL.put(0.76156384f, 35f);
+		CPMULTIPLIER_LEVEL.put(0.764486065f, 35.5f);
+		CPMULTIPLIER_LEVEL.put(0.76739717f, 36f);
+		CPMULTIPLIER_LEVEL.put(0.770297266f, 36.5f);
+		CPMULTIPLIER_LEVEL.put(0.7731865f, 37f);
+		CPMULTIPLIER_LEVEL.put(0.776064962f, 37.5f);
+		CPMULTIPLIER_LEVEL.put(0.77893275f, 38f);
+		CPMULTIPLIER_LEVEL.put(0.781790055f, 38.5f);
+		CPMULTIPLIER_LEVEL.put(0.78463697f, 39f);
+		CPMULTIPLIER_LEVEL.put(0.787473578f, 39.5f);
+		CPMULTIPLIER_LEVEL.put(0.79030001f, 40f);
```

Comment:
You can calculate all of the above values with the forumla listed at the bottom of this article: 
http://pokemongo.gamepress.gg/cp-multiplier

But for the sake of performance and readability I choose to list them all instead of having some complex math in there.